src/AppInstallerRepositoryCore/Microsoft/Schema/1_4/DependenciesTable.cpp

```diff
@@ -0,0 +1,375 @@
+
+    namespace
+    {
+        std::set<Manifest::Dependency> GetDependencies(
+            const Manifest::Manifest& manifest, Manifest::DependencyType dependencyType)
+        {
+            std::set<Manifest::Dependency>  dependencies;
+
+            for (const auto& installer : manifest.Installers)
+            {
+                installer.Dependencies.ApplyToAll([&](Manifest::Dependency dependency)
+                    {
+                        if (dependency.Type == dependencyType)
+                        {
+                            dependencies.emplace(dependency);
+                        }
+                    });
+            }
+
+            return dependencies;
+        }
+
+        void RemoveDependenciesByRowIds(SQLite::Connection& connection, std::vector<SQLite::rowid_t> dependencyRowIds)
+        {
+            using namespace SQLite::Builder;
+            if (dependencyRowIds.empty())
+            {
+                return;
+            }
+            SQLite::Savepoint savepoint = SQLite::Savepoint::Create(connection, std::string{ s_DependenciesTable_Table_Name } + "remove_dependencies_by_rowid");
+
+            SQLite::Builder::StatementBuilder builder;
+            builder.DeleteFrom(s_DependenciesTable_Table_Name).Where(SQLite::RowIDName).In(dependencyRowIds.size());
+
+            SQLite::Statement deleteStmt = builder.Prepare(connection);
+            int bindIndex = 1;
+            for (SQLite::rowid_t rowId : dependencyRowIds)
+            {
+                deleteStmt.Bind(bindIndex, rowId);
+                bindIndex++;
+            }
+
+            deleteStmt.Execute(true);
+
+            savepoint.Commit();
+        }
+
+        void ThrowOnMissingPackageNodes(std::vector<Manifest::Dependency> missingPackageNodes)
+        {
+            if (!missingPackageNodes.empty())
+            {
+                std::string missingPackages{ missingPackageNodes.begin()->Id };
+                std::for_each(
+                    missingPackageNodes.begin() + 1,
+                    missingPackageNodes.end(),
+                    [&](auto& dep) { missingPackages.append(", " + dep.Id);  });
+                THROW_HR_MSG(APPINSTALLER_CLI_ERROR_MISSING_PACKAGE, "Missing packages: %s", missingPackages.c_str());
+            }
+        }
+
+        void InsertManifestDependencies(
+            SQLite::Connection& connection,
+            SQLite::rowid_t manifestRowId,
+            std::set<Manifest::Dependency>& dependencies)
+        {
+            using namespace SQLite::Builder;
+            using namespace Schema::V1_0;
+
+            std::vector<Manifest::Dependency> missingPackageNodes;
+            std::map<Utility::Version, SQLite::rowid_t> versionsMap;
+            std::map<Manifest::Manifest::string_t, SQLite::rowid_t> idsMap;
+            for (const auto& dep : dependencies)
+            {
+                auto depMinVersion = dep.MinVersion.value();
```

Comment:
>dep.MinVersion.value(); [](http://example.com/codeflow?start=37&length=23)

Throws if no MinVersion, which is legitimate.  You don't have any tests for that?